tests/integration/modules/beacons.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# Python Libs
+from __future__ import absolute_import
+import os
+
+# Salt Libs
+from salt.modules import beacons
+from salt.exceptions import CommandExecutionError
+import integration
+
+# Salttesting libs
+from salttesting import skipIf
+from salttesting.helpers import destructiveTest, ensure_in_syspath
+
+ensure_in_syspath('../../')
+
+beacons.__opts__ = {}
+
+BEACON_CONF_DIR = os.path.join(integration.TMP, 'minion.d')
+if not os.path.exists(BEACON_CONF_DIR):
+    os.makedirs(BEACON_CONF_DIR)
+
+
+@skipIf(os.geteuid() != 0, 'You must be root to run these tests')
+@destructiveTest
+class BeaconsAddDeleteTest(integration.ModuleCase):
+    '''
+    Tests the add and delete functions
+    '''
+    def test_add_and_delete(self):
+        '''
+        Test adding and deleting a beacon
+        '''
+        _add = self.run_function('beacons.add', ['ps', {'apache2': 'stopped'}])
+        self.assertTrue(_add['result'])
+
+        # save added beacon
+        _save = self.run_function('beacons.save')
+        self.assertTrue(_save['result'])
+
+        # delete the beacon
+        _delete = self.run_function('beacons.delete', ['ps'])
+        self.assertTrue(_delete['result'])
+
+        # save the results
+        self.run_function('beacons.save')
+
+
+@skipIf(os.geteuid() != 0, 'You must be root to run these tests')
+@destructiveTest
+class BeaconsTest(integration.ModuleCase):
+    '''
+    Tests the beacons execution module
+    '''
+    def setUp(self):
+        try:
+            # Add beacon to disable
+            self.run_function('beacons.add', ['ps', {'apache2': 'stopped'}])
+            self.run_function('beacons.save')
+        except CommandExecutionError:
+            self.skipTest('Unable to add beacon')
+
+    def tearDown(self):
+        # delete added beacon
+        self.run_function('beacons.delete', ['ps'])
+        self.run_function('beacons.save')
+
+    def test_disable(self):
+        '''
+        Test disabling beacons
+        '''
+        ret = self.run_function('beacons.disable')
+        self.assertTrue(ret['result'])
+        # disable added beacon
+        ret = self.run_function('beacons.disable_beacon', ['ps'])
```

Comment:
Better, but what we really want to do I think is to setup up beacons as you have done, assert that they are there by listing the becaons, then disable them and assert that listing them again indicates that they are disabled. Same thing for the other tests in this class.